tests/plugins/widget/manual/partialmask.html

```diff
@@ -0,0 +1,1582 @@
+<div id="editor">
+	<p>foo</p>
+	<figure class="image"><img alt="" height="100" src="%BASE_PATH%/_assets/lena.jpg" width="100" />
+		<figcaption>Caption</figcaption>
+	</figure>
+	<p>bar</p>
+	<div class="partialMasked"><div class="notmasked">You can type here...</div><div class="content">...but not here</div></div>
+</div>
+
+<script>
+	// Ignore on mobiles.
+	if ( bender.tools.env.mobile ) {
+		bender.ignore();
+	}
+	var editor = CKEDITOR.replace( 'editor', {
+			extraPlugins: 'partialMasked,image3',
+			height: 500
+		} ),
+		info = document.getElementById( 'info' );
+
+	CKEDITOR.plugins.add( 'partialMasked', {
+		requires: 'widget',
+		init: function( editor ) {
+			editor.widgets.add( 'partialMasked', {
+				button: 'Partial mask widget',
+				pathName: 'test-widget',
+
+				template:
+					'<div class="partialMasked">' +
+						'<div class="notmasked"></div>' +
+						'<div class="content"></div>' +
+					'</div>',
+
+				editables: {
+					nomask: {
+						selector: '.notmasked',
+						allowedContent: 'br'
+					},
+					content: {
+						selector: '.content',
+						allowedContent: 'br'
+					}
+				},
+
+				parts: {
+					nomask: '.notmasked',
+					content: '.content'
+				},
+
+				allowedContent: 'div(partialMasked,content,notmasked)',
+				requiredContent: 'div(partialMasked)',
+
+				mask: 'content',
+
+				upcast: function( element ) {
+					return element.name == 'div' && element.hasClass( 'partialMasked' );
+				}
+			} );
+		}
+	} );
+
+	// 'image3' plugin is added to replicate the behaviour of widget with resizer.
+	// It's basically 'image2' plugin with partial mask on 'caption' and some stuff
+	// like dialog and context menu removed.
+	// Please note that 'image2' plugin is still being loaded in .md file to have lang files etc.
+
+	var template = '<img alt="" src="" />',
+		templateBlock = new CKEDITOR.template(
+			'<figure class="{captionedClass}">' +
+				template +
+				'<figcaption>{captionPlaceholder}</figcaption>' +
+			'</figure>' ),
+		alignmentsObj = { left: 0, center: 1, right: 2 },
+		regexPercent = /^\s*(\d+\%)\s*$/i;
+
+	CKEDITOR.plugins.add( 'image3', {
```

Comment:
That's a lot of duplication, maybe there is some simple way (even hacky) to overwrite `image2` widget definition to change it's mask property so the partial mask is applied? 🤔  I would leave dialog and context menu untouched (so one can test e.g. changing widget size by dialog).